community/kernel/src/main/java/org/neo4j/graphdb/GraphDatabaseService.java

```diff
@@ -183,10 +187,55 @@
      * space).
      *
      * @return all relationship types in the underlying store
-     * @deprecated this operation can be found in {@link GlobalGraphOperations} instead.
      */
-    @Deprecated
-    Iterable<RelationshipType> getRelationshipTypes();
+    ResourceIterable<RelationshipType> getRelationshipTypes();
+
+    /**
+     * Returns all relationship types currently in use in the underlying store. Relationship types are
+     * added to the underlying store the first time they are used in a successfully committed
+     * {@link Node#createRelationshipTo node.createRelationshipTo(...)}. Note that this method is
+     * guaranteed to return all known relationship types, and it guarantees that it won't return
+     * "historic" relationship types that no longer have any relationships in the graph.
+     *
+     * @return all relationship types in use in the underlying store
+     */
+    ResourceIterable<RelationshipType> getAllRelationshipTypesInUse();
```

Comment:
So like this:

* `getRelationshipTypes()` - returning the relationship types that are in use _(noting that this method exists from before, and this behaviour is a change from previous behaviour)_, and
* `getAllRelationshipTypes()` - returning all the relationship types, even the ones that are not in use _(where it is up for debate wether we should have this method at all, since the unused types are pretty useless)_.

Did I understand that correctly, @tinwelint?